app/models/katello/content_view_version.rb

```diff
@@ -240,12 +240,22 @@ def check_ready_to_promote!
     end
 
     def check_ready_to_destroy!
+      check_used_by_composite!
+
       if environments.any? && !organization.being_deleted?
         fail _("Cannot delete version while it is in environments: %s") % environments.map(&:name).join(",")
       end
       return true
     end
 
+    def check_used_by_composite!
+      if composite_content_views.any? && !organization.being_deleted?
+        fail _("Cannot delete version while it is in use by composite content views: %s") %
```

Comment:
I wouldn't expect a 'check_used_by_composite!' function to throw me failures about not being able to delete a version.